document/core/exec/runtime.rst

```diff
@@ -678,51 +697,56 @@ This definition allows to index active labels surrounding a :ref:`branch <syntax
    The selected label is identified through the :ref:`label index <syntax-labelidx>` :math:`l`, which corresponds to the number of surrounding |LABEL| instructions that must be hopped over -- which is exactly the count encoded in the index of a block context.
 
 
-.. index:: ! throw context, exception, throw address, catch block
+.. index:: ! throw context, tag, throw address, catch block, handler, exception
 .. _syntax-ctxt-throw:
 
 Throw Contexts
 ..............
 
 In order to specify the reduction of |TRY| blocks
-with the help of the administrative instructions |THROWADDR|, |REFEXNADDR|, and |CATCHN|,
+with the help of the administrative instructions |THROWadm|, |CATCHadm|, |DELEGATEadm|, and |CAUGHTadm|,
 the following syntax of *throw contexts* is defined, as well as associated structural rules:
 
 .. math::
    \begin{array}{llll}
    \production{(throw contexts)} & \XT &::=&
      \val^\ast~[\_]~\instr^\ast \\ &&|&
      \LABEL_n\{\instr^\ast\}~\XT~\END \\ &&|&
-     \CATCHN_n\{\instr^\ast\}~\XT~\END \\ &&|&
+     \CAUGHTadm\{a~\val^\ast\}~\XT~\END \\ &&|&
      \FRAME_n\{F\}~\XT~\END \\
    \end{array}
 
-Throw contexts allow matching the program context around a throw instruction up to the nearest enclosing :math:`\CATCHN` handler, thereby selecting the exception handler responsible for an exception.
+Throw contexts allow matching the program context around a throw instruction up to the innermost enclosing |CATCHadm| or |DELEGATEadm|, thereby selecting the exception |handler| responsible for an exception, if one exists.
+
+.. note::
+   Contrary to block contexts, throw contexts don't skip over handlers.
+
+   Since handlers are not included above, there is always a unique maximal throw context to match the reduction rules.
+   This basically means that :math:`\CAUGHTadm {..} instr^\ast \END` is not a potential catching block for exceptions thrown by :math:`\instr^\ast`, since it is inside a |handler|'s block.
 
 .. note::
-   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try>` would be as follows.
+   For example, catching a simple :ref:`throw <exec-throw>` in a :ref:`try block <exec-try-catch>` would be as follows.
```

Comment:
Adding a `..todo::' to include a more complex example using a throw context other then `T=[_]`. 

Question for the next PR, would you prefer to change the current simple example to the new more complex example, or to add the more complex example in addition to the simple one?

Reply:
I think the current example is complicated enough, and actually I don't think I understand it 😂
But for the sake of resolving as many threads as possible, let's resolve this because this is talking about whether the current example is sophisticated enough. I'll ask the question about the example elsewhere.